en/ros2/px4_sdk.md

```diff
@@ -0,0 +1,336 @@
+# PX4 SDK
+
+The [PX4 SDK](https://github.com/PX4/px4_sdk) is a C++ library to simplify controlling PX4 from ROS 2.
+It allows to write external modes that are dynamically registered with PX4 and behave the same way as internal ones.
+A mode can send different types of setpoints, ranging from high-level navigation tasks all the way down to direct actuator controls.
+
+## Overview
+
+This diagram provides a conceptual overview:
+
+![ROS2 modes overview diagram](../../assets/middleware/ros2/px4_sdk/ros2_modes_overview.svg)
+
+<!-- Source: https://docs.google.com/drawings/d/1WByCfgcytnaow7r41VhYJL8OGrw1RjFO51GoPMQBCNA/edit -->
+
+The following sections define and explain the terms used in the diagram.
+
+### Definitions
+
+#### Mode
+
+- A mode is a component that can send setpoints to the vehicle, which control its motion (such as velocity or direct actuator commands).
+- It is passive, meaning it does not activate other modes (by sending commands), and it must be activated by someone else (either the user through RC/GCS, the FMU in a failsafe situation or by a _mode executor_).
+- It selects a setpoint type and sends it while it is active.
+  A mode can switch between multiple setpoint types.
+- Has a name displayed by the GCS.
+- Can configure its mode requirements (for example that it requires a valid position estimate).
+- A mode can consist of different tasks (for example flying to a target, lowering a winch, releasing a payload and then fly back).
+
+#### Mode Executor
+
+- A mode executor is an optional component one level higher than a mode. It is a state machine that can activate modes, and wait for their completion.
+- It can only do so while it is in charge.
+  For that, an executor has exactly one _owned mode_ (and a mode can be owned by at most one executor).
+  This mode serves as activation for the executor: when the user selects the mode, the owning executor gets activated and can select any mode.
+  It stays in charge until the user switches modes (by RC or from a GCS), or a failsafe triggers a mode switch.
+  Should the failsafe clear, the executor gets reactivated.
+- This allows multiple executors to coexist.
+- Executors cannot activate other executors.
+- Within the SDK, a mode executor is always implemented in combination with a custom mode.
+
+:::note
+- These definitions guarantee that a user can take away control from a custom mode or executor at any point in time by commanding a mode switch through RC or a GCS.
+- A mode executor is transparent to the user.
+  It gets indirectly selected and activated through the owning mode, and thus the mode should be named accordingly.
+:::
+
+##### When would you use a mode executor?
+
+The most common use-case is the execution of an autonomous 'mission', such as a survey or a payload delivery: the executor triggers a take-off, then switches to a custom mode which performs the survey, and when completed, triggers an RTL.
```

Comment:
As above, I think this would be better integrated into the topic https://github.com/PX4/PX4-user_guide/pull/2740/files#r1332395891